src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitTransactionDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for(FruitTransaction transaction : transactions) {
+            OperationHandler operationHandler = operationStrategy.getOperationHandler(transaction.getOperation());
+            operationHandler.recount(transaction);
+        }
+    }
+}
```

Comment:
It seems like the `ShopServiceImpl` class is directly tied to the `OperationStrategy` which makes it less independent. Consider making your services more independent and calling them in the `main()` method, where the result of one service method can be passed as an input to the next one.